src/Npgsql/TypeHandling/DefaultTypeHandlerFactory.cs

```diff
@@ -15,7 +15,7 @@ internal DefaultTypeHandlerFactory(Type handlerType)
         {
             // Recursively look for the TypeHandler<T> superclass to extract its T as the
             // DefaultValueType
-            var baseClass = handlerType;
+            Type? baseClass = handlerType;
```

Comment:
Why is this needed? Also, does the next line compile (we call GetTypeInfo without a check...)

Reply:
`BaseClass` which used later can be null. See https://github.com/dotnet/coreclr/blob/45851dffd1f6dd1550a9be691247326090dca8c9/src/System.Private.CoreLib/shared/System/Type.cs#L299.

Reply:
Ah of course, didn't look at the context...